src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -657,32 +657,42 @@ public String toString() {
      *
      * @see StoreFileMetaData
      */
-    public final static class MetadataSnapshot implements Iterable<StoreFileMetaData>, Streamable {
+    public final static class MetadataSnapshot implements Iterable<StoreFileMetaData> {
```

Comment:
we have a new interface `Writeable.java` that we can use here. It has:

```Java
    void writeTo(StreamOutput out) throws IOException;
    T readFrom(StreamInput in) throws IOException;
```
and it's usually implemented by passing the `StreamInput` to a dedicated ctor so everything can be final. Can we use that here too?